drivers/SmartThings/matter-switch/src/aqara-cube/init.lua

```diff
@@ -0,0 +1,224 @@
+
+local function endpoint_to_component(device, endpoint)
+  return "main"
+end
+
+local function device_init(driver, device)
+  if device.network_type == device_lib.NETWORK_TYPE_MATTER then
+    device:subscribe()
+    device:set_endpoint_to_component_fn(endpoint_to_component)
+  end
+end
+
+-- This is called either on add for parent/child devices, or after the device profile changes for components
+local function configure_buttons(device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.SwitchFeature.MOMENTARY_SWITCH})
+    device.log.debug(#MS.." momentary switch endpoints")
+    for _, ep in ipairs(MS) do
+      -- device only supports momentary switch, no release events
+      device.log.debug("configuring for press event only")
+      set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, true)
+    end
+  end
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+    local MS = get_reordered_endpoints(driver, device)
+    local main_endpoint = device.MATTER_DEFAULT_ENDPOINT
+    if #MS > 0 then
+      main_endpoint = MS[1] -- the endpoint matching to the non-child device
+      if MS[1] == 0 then main_endpoint = MS[2] end -- we shouldn't hit this, but just in case
+    end
+    device.log.debug("main button endpoint is "..main_endpoint)
+
+    -- At the moment, we're taking it for granted that all momentary switches only have 2 positions
+    local current_component_number = 1
+    local component_map = {}
+    for _, ep in ipairs(MS) do -- for each momentary switch endpoint (including main)
+      device.log.debug("Configuring endpoint "..ep)
+      -- build the mapping of endpoints to components
+      component_map[string.format("%d", current_component_number)] = ep
+      current_component_number = current_component_number + 1
+    end
+
+    device:set_field(COMPONENT_TO_ENDPOINT_MAP_BUTTON, component_map, {persist = true})
+    device:try_update_metadata({profile = "cube-t1-pro"})
+    device:set_field(DEFERRED_CONFIGURE, true)
+  end
+end
+
+local function info_changed(driver, device, event, args)
+  if device.profile.id ~= args.old_st_store.profile.id
+    and device:get_field(DEFERRED_CONFIGURE)
+    and device.network_type ~= device_lib.NETWORK_TYPE_CHILD then
+
+    -- At the time of device_added, there is an error that the corresponding capability cannot be found
+    -- because the profile has not been changed from the generic profile of fingerprint to the Aqara Cube.
+    reset_thread(device)
+    device:emit_event(cubeFace.cubeFace("face1Up"))
+
+    device:set_field(DEFERRED_CONFIGURE, nil)
+
+    -- In the current test framework, the values of device.profile.id and args.old_st_store.profile.id are always the same,
+    -- so the test coverage cannot be increased. This is why I added DOING_CONFIGURE flag.
+    device:set_field(DOING_CONFIGURE, true)
```

Comment:
@nickolas-deboom @hcarter-775 Hi, 
Although already described in the inline comments, I could not find a way to test the code in case of DEFERRED_CONFIGURE in the current test framework. This is because the values of `device.profile.id` and `args.old_st_store.profile.id` are always equal to `00000000-1111-2222-3333-0000000002`. Therefore, I added a variable called `DOING_CONFIGURE`, which can be changed immediately in the test file, but the test case is intermittently failed because the capability information is not updated in time when calling emit_event . Is there any way to solve it?

Reply:
Yes, we have a function to solve this called `test.wait_for_events()` that should fix the intermittent failing. Simply place this function before the event that would need the capability information to be updated and this _should_ fix it, based on your problem description.

Reply:
Yes, I've already used that method. However, as you said, I can always succeed if I call as below, but I can not satisfy test coverage. This is because the `received lifecycle event: infoChanged` has already ended, so the `mock_device:set_field (DOING_CONFIGURE, true)` that is called after that has no meaning.
>test.socket.device_lifecycle:__queue_receive(mock_device:generate_info_changed({value = "face1Up"}))
>test.wait_for_events()

Or when I call as below, the test case still fails intermittently as mentioned above.
>test.socket.device_lifecycle:__queue_receive(mock_device:generate_info_changed({value = "face1Up"}))
>mock_device:set_field(DOING_CONFIGURE, true)
>test.wait_for_events()

Reply:
@hcarter-775 Thank you very much. According to your guide, I add the contents below,
>test.add_package_capability("cubeAction.yml")
>test.add_package_capability("cubeFace.yml")
And I was able to raise the test coverage to 91% by `setting_field` before `info_changed` as below.
>mock_device:set_field(DOING_CONFIGURE, true)
>test.socket.device_lifecycle:__queue_receive(mock_device:generate_info_changed({value = "face1Up"}))
> test.wait_for_events()